x-pack/protocol/src/main/java/org/elasticsearch/protocol/xpack/ml/datafeed/DatafeedUpdate.java

```diff
@@ -0,0 +1,330 @@
+        if (scriptFields != null) {
+            builder.startObject(DatafeedConfig.SCRIPT_FIELDS.getPreferredName());
+            for (SearchSourceBuilder.ScriptField scriptField : scriptFields) {
+                scriptField.toXContent(builder, params);
+            }
+            builder.endObject();
+        }
+        addOptionalField(builder, DatafeedConfig.SCROLL_SIZE, scrollSize);
+        addOptionalField(builder, DatafeedConfig.CHUNKING_CONFIG, chunkingConfig);
+        builder.endObject();
+        return builder;
+    }
+
+    private void addOptionalField(XContentBuilder builder, ParseField field, Object value) throws IOException {
+        if (value != null) {
+            builder.field(field.getPreferredName(), value);
+        }
+    }
+
+    String getJobId() {
+        return jobId;
+    }
+
+    TimeValue getQueryDelay() {
+        return queryDelay;
+    }
+
+    TimeValue getFrequency() {
+        return frequency;
+    }
+
+    List<String> getIndices() {
+        return indices;
+    }
+
+    List<String> getTypes() {
+        return types;
+    }
+
+    Integer getScrollSize() {
+        return scrollSize;
+    }
+
+    QueryBuilder getQuery() {
+        return query;
+    }
+
+    AggregatorFactories.Builder getAggregations() {
+        return aggregations;
+    }
+
+    List<SearchSourceBuilder.ScriptField> getScriptFields() {
+        return scriptFields == null ? Collections.emptyList() : scriptFields;
+    }
+
+    ChunkingConfig getChunkingConfig() {
+        return chunkingConfig;
+    }
+
+    /**
+     * The lists of indices and types are compared for equality but they are not
+     * sorted first so this test could fail simply because the indices and types
+     * lists are in different orders.
+     */
+    @Override
+    public boolean equals(Object other) {
+        if (this == other) {
+            return true;
+        }
+
+        if (other == null || getClass() != other.getClass()) {
+            return false;
+        }
+
+        DatafeedUpdate that = (DatafeedUpdate) other;
+
+        return Objects.equals(this.id, that.id)
+            && Objects.equals(this.jobId, that.jobId)
+            && Objects.equals(this.frequency, that.frequency)
+            && Objects.equals(this.queryDelay, that.queryDelay)
+            && Objects.equals(this.indices, that.indices)
+            && Objects.equals(this.types, that.types)
+            && Objects.equals(this.query, that.query)
+            && Objects.equals(this.scrollSize, that.scrollSize)
+            && Objects.equals(this.aggregations, that.aggregations)
+            && Objects.equals(this.scriptFields, that.scriptFields)
+            && Objects.equals(this.chunkingConfig, that.chunkingConfig);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(id, jobId, frequency, queryDelay, indices, types, query, scrollSize, aggregations, scriptFields,
+            chunkingConfig);
+    }
+
+    boolean isNoop(DatafeedConfig datafeed) {
```

Comment:
I don't think this method is needed in the client.  (I think it was only added in the server to be used by some tests.)